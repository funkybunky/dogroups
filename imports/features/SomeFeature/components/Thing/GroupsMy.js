/* global Meteor */
import React from 'react';
import GroupsList from './ListGroups';

const GroupsMy = (props) => {
  const userGroupIds = props.user.myGroups;
  if (!userGroupIds) throw new Error('NOOOOO myGroups prop not on user obj');
  const userGroups = userGroupIds.reduce((myGroups, groupId) => {
    const groups = props.groups.filter((group) => group._id === groupId);
    if (groups.length === 1) {
      return myGroups.concat(groups);
    }
    return myGroups;
  }, []);
  return (
    <GroupsList
      groups={userGroups}
      joinGroup={props.joinGroup}
      title="My Groups"
    />
  );
};

GroupsMy.propTypes = {
  groups: React.PropTypes.array,
  joinGroup: React.PropTypes.func,
  user: React.PropTypes.object,
};

export default GroupsMy;
