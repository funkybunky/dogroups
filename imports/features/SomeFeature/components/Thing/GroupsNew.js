import React, { Component } from 'react';
import Radium from 'radium';
import { browserHistory } from 'react-router';

import { Bert } from 'meteor/themeteorchef:bert';
import Form from 'react-jsonschema-form';

const schema = {
  title: 'Create new Group',
  type: 'object',
  required: ['name'],
  properties: {
    name: { type: 'string', title: 'Group Name', default: 'Yogi Masters' },
  },
};

const uiSchema = {
};

const log = (type) => console.log.bind(console, type);

class GroupsNew extends Component {

  handleSubmit = ({ formData }) => {
    this.props.createGroup(formData, (err) => {
      if (err) {
        log('error')(err);
        Bert.alert(err.reason, 'danger');
      } else {
        Bert.alert(`Successfully created group ${formData.name}`, 'success');
        browserHistory.push('/groups');
        // TODO: redirect to group detail page, right?
      }
    });
  }

  render() {
    return (
      <Form
        schema={schema}
        uiSchema={uiSchema}
        onChange={log('changed')}
        onSubmit={this.handleSubmit}
        onError={log('errors')}
      />
    );
  }
}

// const styles = {
//   root: {
//     display: 'inline',
//   },
//   button: {
//     background: '#eee',
//     border: '1px solid #ddd',
//     fontSize: 24,
//   },
// };

GroupsNew.propTypes = {
  groups: React.PropTypes.array, // are needed to check for existing ones
  createGroup: React.PropTypes.func, // callback to method
};

export default Radium(GroupsNew);
