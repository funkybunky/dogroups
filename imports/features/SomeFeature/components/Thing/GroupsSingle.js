import React from 'react';
import Radium from 'radium';

const GroupsSingle = (props) => {
  console.log('whuu');
  const groupId = props.params.id;
  const group = props.groups.filter((group) => group._id === groupId)[0];
  if (!group) console.log('nooo! param ID doesnt match any group. redirect!');
  return (
    <div>
      <h2>{group.name}</h2>
    </div>
  );
};

const styles = {
  root: {
    display: 'inline',
  },
  button: {
    background: '#eee',
    border: '1px solid #ddd',
    fontSize: 24,
  },
};

GroupsSingle.propTypes = {
  groups: React.PropTypes.array,
};

export default Radium(GroupsSingle);
