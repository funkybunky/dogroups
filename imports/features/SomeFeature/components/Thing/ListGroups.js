import React, { Component } from 'react';
import Radium from 'radium';

class GroupElem extends Component {
  handleJoinClick = (event) => {
    event.preventDefault();
    this.props.joinGroup(this.props.id);
  }
  render() {
    return (
      <div>
        <a href={`/groups/${this.props.id}`}><h2>{this.props.name}</h2></a>
        <button onClick={this.handleJoinClick}>Join</button>
        <hr />
      </div>
    );
  }
}

// TODO: see if you can change it to a stateless component, works with Radium?
class ListGroups extends Component {

  render() {
    return (
      <div>
        <h2>{this.props.title ? this.props.title : 'All Groups'}:</h2>
        {this.props.groups.map((group) =>
          <GroupElem
            name={group.name}
            key={group._id}
            id={group._id}
            joinGroup={this.props.joinGroup}
          />
        )}
      </div>
    );
  }
}

const styles = {
  root: {
    display: 'inline',
  },
  button: {
    background: '#eee',
    border: '1px solid #ddd',
    fontSize: 24,
  },
};

ListGroups.propTypes = {
  groups: React.PropTypes.array,
  joinGroup: React.PropTypes.func,
  title: React.PropTypes.string,
};

export default Radium(ListGroups);
