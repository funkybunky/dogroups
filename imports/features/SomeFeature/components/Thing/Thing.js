import React, { Component } from 'react';
import Radium from 'radium';

class Thing extends Component {
  callMethod = () => {
    this.props.createGroup({ name: 'Unknown' }, (error, result) => {
      console.log(result);
    });
  };

  showProps = () => {
    console.log(this.props);
  };

  render() {
    return (
      <div style={styles.root}>
        <div>
          <button
            style={styles.button}
            onClick={this.showProps}
          >
            Dump props
          </button>
        </div>
        <div>
          <button
            style={styles.button}
            onClick={this.callMethod}
          >
            Call Meteor method
          </button>
        </div>
      </div>
    );
  }
}

const styles = {
  root: {
    display: 'inline',
  },
  button: {
    background: '#eee',
    border: '1px solid #ddd',
    fontSize: 24,
  },
};

Thing.propTypes = {
  ready: React.PropTypes.bool.isRequired,
  createGroup: React.PropTypes.object.isRequired,
  groups: React.PropTypes.array.isRequired,
};

export default Radium(Thing);
