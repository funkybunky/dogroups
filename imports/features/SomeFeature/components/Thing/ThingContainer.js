import { Meteor } from 'meteor/meteor';
import { Groups } from 'imports/collections/groups.js';
import { createContainer } from 'meteor/react-meteor-data';
import { createGroup as _createGroup } from '../../methods';
import joinGroup from 'imports/features/SomeFeature/methods/joinGroup.method';

import Thing from './Thing';
export default createContainer(() => {
  const groupsHandle = Meteor.subscribe('groups.all');

  // Maybe write a higher-order function for this later
  const createGroup = (args, callback) => {
    _createGroup.call(args, callback);
  };

  const joinGroupHandler = (args, cb) => {
    joinGroup.call(args, cb);
  };

  return {
    ready: groupsHandle.ready(),
    groups: Groups.find({}).fetch(),
    createGroup,
    joinGroup: joinGroupHandler,
  };
}, Thing);
