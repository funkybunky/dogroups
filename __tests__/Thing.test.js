import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import Thing from '../imports/features/SomeFeature/components/Thing/Thing';

test('Just takes a snapshot', () => {
  const component = renderer.create(
    <Thing />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
