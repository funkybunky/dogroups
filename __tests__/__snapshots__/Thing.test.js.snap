exports[`test whoa 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "display": "inline"
    }
  }>
  <button
    data-radium={true}
    onClick={[Function anonymous]}
    style={
      Object {
        "background": "#eee",
        "border": "1px solid #ddd",
        "fontSize": 24
      }
    }>
    Dump props
  </button>
</div>
`;
